### PR TITLE
Migration Python 2 to 3

### PR DESCRIPTION
**Pull request by adityacp**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,12 +10,9 @@ env:
 
 # command to install dependencies
 install:
-  - pip install tornado
-  - pip install git+https://github.com/FOSSEE/online_test.git#egg=yaksh-0.1
+  - if [[$TRAVIS_PYTHON_VERSION == 2.7 ]]; then pip install -r requirements/requirements-py2.txt; fi
```
Comment (prabhuramachandran) [inline .travis.yml]: I generally use explicit string comparisons "2.7".

Comment (adityacp) [inline .travis.yml]: This has been removed and handled in setup.py

File: yaksh/cpp_stdio_evaluator.py [hunk 2]
```diff
@@ -17,13 +17,13 @@ def setup(self):
...
     def teardown(self):
-        super(CppStdioEvaluator, self).teardown()
         os.remove(self.submit_code_path)
         if self.files:
             delete_files(self.files)
+        super(CppStdioEvaluator, self).teardown()
 
     def set_file_paths(self):
-        user_output_path = os.getcwd() + '/output'
+        user_output_path = os.getcwd() + '/output_file'
```
Comment (prabhuramachandran) [inline yaksh/cpp_stdio_evaluator.py]: Please use `os.path.join`.

File: yaksh/evaluator_tests/test_bash_evaluation.py [hunk 3]
```diff
@@ -8,15 +11,23 @@
 
 class BashAssertionEvaluationTestCases(unittest.TestCase):
     def setUp(self):
+        with open('/tmp/test.txt', 'wb') as f:
```
Comment (prabhuramachandran) [inline yaksh/evaluator_tests/test_bash_evaluation.py]: Use tempfile.

File: yaksh/evaluator_tests/test_java_evaluation.py [hunk 4]
```diff
@@ -97,20 +106,26 @@ class Test{
...
                                'expected_input': '5\n6'}]
         evaluator.SERVER_TIMEOUT = 4
         self.timeout_msg = ("Code took more than {0} seconds to run. "
                             "You probably have an infinite loop in"
                             " your code.").format(evaluator.SERVER_TIMEOUT)
 
-    def teardown(self):
+    def tearDown(self):
         evaluator.SERVER_TIMEOUT = 4
+        os.remove('/tmp/test.txt')
```
Comment (prabhuramachandran) [inline yaksh/evaluator_tests/test_java_evaluation.py]: tempfile.

File: yaksh/evaluator_tests/test_java_evaluation.py [hunk 5]
```diff
@@ -234,13 +250,45 @@ class Test
...
         kwargs = {'user_answer': user_answer,
                   'test_case_data': self.test_case_data
                   }
         result = get_class.evaluate(**kwargs)
-        self.assertEquals(result.get('error'), "Correct answer")
+        self.assertEqual(result.get('error'), "Correct answer")
         self.assertTrue(result.get('success'))
 
+    def test_file_based_stdout(self):
+        self.file_paths = [("/tmp/test.txt", False)]
```
Comment (prabhuramachandran) [inline yaksh/evaluator_tests/test_java_evaluation.py]: tempfile.

File: yaksh/java_stdio_evaluator.py [hunk 6]
```diff
@@ -1,11 +1,12 @@
 #!/usr/bin/env python
+from __future__ import unicode_literals
 import subprocess
 import os
 from os.path import isfile
 
-#local imports
-from stdio_evaluator import StdIOEvaluator
-from file_utils import copy_files, delete_files
+#Local imports
```
Comment (prabhuramachandran) [inline yaksh/java_stdio_evaluator.py]: PEP 8.

**Comment by prabhuramachandran** [comment]
Merging for now.

**Pull request merged**